internal/chain/withdraw_tokens.go

```diff
@@ -53,56 +44,41 @@ func checkTokenTx(st *StateManager, tx *transactions.GenTransaction) ([]*tokenSe
 	case *protocol.AnonTokenAccount:
 		account = sponsor
 	default:
-		return nil, nil, nil, nil, fmt.Errorf("%v cannot sponsor token transactions", st.Sponsor.Header().Type)
+		return fmt.Errorf("%v cannot sponsor token transactions", st.Sponsor.Header().Type)
 	}
 
 	tokenUrl, err := account.ParseTokenUrl()
 	if err != nil {
-		return nil, nil, nil, nil, fmt.Errorf("invalid token URL: %v", err)
+		return fmt.Errorf("invalid token URL: %v", err)
 	}
 
 	//now check to see if we can transact
 	//really only need to provide one input...
 	//now check to see if the account is good to send tokens from
-	amt := types.Amount{}
-	txAmt := big.NewInt(0)
-	for _, val := range body.To {
-		amt.Add(amt.AsBigInt(), txAmt.SetUint64(val.Amount))
-	}
-
-	if !account.CanDebitTokens(&amt.Int) {
-		return nil, nil, nil, nil, fmt.Errorf("insufficient balance")
+	total := types.Amount{}
+	for _, to := range body.To {
+		total.Add(total.AsBigInt(), new(big.Int).SetUint64(to.Amount))
```

Comment:
This isn't a problem we need to tackle now, so this is food for thought.  The "amount" is a uint64. If we support up to 18 decimals places of precision similar to the erc20 spec, then using unit64 will be extremely limiting for transactions.  There are uint128 implementations but those will greatly complicate marshaling in other languages.

Reply:
https://accumulate.atlassian.net/browse/AC-555